prow/plugins/lgtm/lgtm.go

```diff
@@ -74,18 +76,22 @@ func handle(gc githubClient, log *logrus.Entry, ic github.IssueCommentEvent) err
 	isAssignee := ic.Issue.IsAssignee(commentAuthor)
 	isAuthor := ic.Issue.IsAuthor(commentAuthor)
 	if isAuthor && wantLGTM {
-		resp := "you can't LGTM your own PR"
+		resp := "you cannot LGTM your own PR"
 		log.Infof("Commenting with \"%s\".", resp)
 		return gc.CreateComment(org, repo, number, plugins.FormatResponse(ic.Comment, resp))
 	} else if !isAuthor {
-		if !isAssignee && wantLGTM {
-			resp := "you can't LGTM a PR unless you are an assignee"
-			log.Infof("Commenting with \"%s\".", resp)
-			return gc.CreateComment(org, repo, number, plugins.FormatResponse(ic.Comment, resp))
-		} else if !isAssignee && !wantLGTM {
-			resp := "you can't remove LGTM from a PR unless you are an assignee"
-			log.Infof("Commenting with \"%s\".", resp)
-			return gc.CreateComment(org, repo, number, plugins.FormatResponse(ic.Comment, resp))
+		if !isAssignee {
+			log.Infof("Assigning %s/%s#%d to %s", org, repo, number, commentAuthor)
+			if err := gc.AssignIssue(org, repo, number, []string{commentAuthor}); err != nil {
+				msg := "assigning you to the PR failed"
+				if ok, err := gc.IsMember(org, commentAuthor); err == nil && !ok {
```

Comment:
If err is not nil then log it at least. This code is now indented several times and a little confusing to follow.